openproblems/tasks/chromatin_potential/methods/linear_regression.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+import scanpy as sc
+import subprocess
+from ....tools.decorators import method
+
+
+def _chrom_limit(x, tss_size=2e5):
+    y = x.values
+    gene_direction = y[-1]
+    gene_start = y[-3]
+    gene_end = y[-2]
+    if gene_direction == "+":
+        return [gene_start - tss_size // 2, gene_start + tss_size // 2]
+    else:
+        return [gene_end - tss_size // 2, gene_end + tss_size // 2]
+
+
+def _get_annotation(adata):
+    from pyensembl import EnsemblRelease
+
+    # TODO: add more species
+    subprocess.call(
+        ["pyensembl", "install", "--release", "100", "--species", "mus_musculus"]
+    )
+    data = EnsemblRelease(100, species="mus_musculus")
+    # get ensemble gene coordinate
+    genes = []
+    for i in adata.var.index.map(lambda x: x.split(".")[0]):
+        try:
+            gene = data.gene_by_id(i)
+            genes.append(
+                [
+                    "chr%s" % gene.contig,
+                    gene.start,
+                    gene.end,
+                    gene.strand,
+                ]
+            )
+        except:
+            genes.append([np.nan, np.nan, np.nan, np.nan])
+    old_col = adata.var.columns.values
+    adata._var = pd.concat(
+        [adata.var, pd.DataFrame(genes, index=adata.var_names)], axis=1
+    )
+    adata._var.columns = np.hstack(
```

Comment:
Why are we putting this in `_var`? We should only use the public interface of `AnnData`.

Reply:
I'll fix it with .var.